ext/curl/interface.c

```diff
@@ -2777,13 +2815,21 @@ static int _php_curl_setopt(php_curl *ch, zend_long option, zval *zvalue) /* {{{
 							}
 
 #if LIBCURL_VERSION_NUM >= 0x073800 /* 7.56.0 */
+							if (!(stream = php_stream_open_wrapper(ZSTR_VAL(postval), "rb", STREAM_MUST_SEEK, NULL))) {
+								zend_string_release_ex(string_key, 0);
+								return FAILURE;
+							}
+							if (php_stream_stat(stream, &ssb)) {
+								zend_string_release_ex(string_key, 0);
+								return FAILURE;
+							}
```

Comment:
It might be worth to support non seekable stream, like URL, which is a completely legit scenario.

Thanks.

Reply:
That may not always work (see https://curl.haxx.se/libcurl/c/curl_mime_data_cb.html), but we still could fail in this case.

Reply:
Passing -1 for data length should be ok in general and would spare the stat call. Not requiring stream to be seekable and setting the seek function to NULL might be still a fallback. Knowing the stream size ahead is good, but doesn't guarantee we still don't end up with a partial data as it's stream dependent.

Thanks.

Reply:
The [curl docs](https://curl.haxx.se/libcurl/c/curl_mime_data_cb.html) are not particularly clear regarding the `datasize`, but it seems passing `-1` works.

Instead of passing `NULL` as `seekfunc` I'd suggest to have a callback which tries to seek, and returns `CURL_SEEKFUNC_CANTSEEK` otherwise. which indicates that "libcurl is free to work around the problem if possible".

Reply:
Yep, could be probably an option, too. Thanks.